service/tracker/consensus_test.go

```diff
@@ -0,0 +1,908 @@
+		)
+
+		got, err := cons.Commit(header.Height)
+
+		require.NoError(t, err)
+		assert.Equal(t, blockData.FinalStateCommitment, got)
+	})
+
+	t.Run("handles requested height over last finalized height", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(header.Height, header.ID())))
+
+		holder := mocks.BaselineRecordHolder(t)
+		holder.RecordFunc = func(flow.Identifier) (*uploader.BlockData, error) {
+			return blockData, nil
+		}
+
+		cons := tracker.BaselineConsensus(
+			t,
+			tracker.WithDB(db),
+			tracker.WithLast(header.Height),
+			tracker.WithHolder(holder),
+		)
+
+		_, err := cons.Commit(header.Height + 999)
+
+		assert.Error(t, err)
+	})
+
+	t.Run("handles missing block height index in DB", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		// Missing block index in database.
+		//require.NoError(t, db.Update(operation.IndexBlockHeight(header.Height, header.ID())))
+
+		holder := mocks.BaselineRecordHolder(t)
+		holder.RecordFunc = func(flow.Identifier) (*uploader.BlockData, error) {
+			return blockData, nil
+		}
+
+		cons := tracker.BaselineConsensus(
+			t,
+			tracker.WithDB(db),
+			tracker.WithLast(header.Height),
+			tracker.WithHolder(holder),
+		)
+
+		_, err := cons.Commit(header.Height)
+
+		assert.Error(t, err)
+	})
+
+	t.Run("handles record holder failure", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(header.Height, header.ID())))
+
+		holder := mocks.BaselineRecordHolder(t)
+		holder.RecordFunc = func(flow.Identifier) (*uploader.BlockData, error) {
+			return nil, mocks.GenericError
+		}
+
+		cons := tracker.BaselineConsensus(
+			t,
+			tracker.WithDB(db),
+			tracker.WithLast(header.Height),
+			tracker.WithHolder(holder),
+		)
+
+		_, err := cons.Commit(header.Height)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestConsensus_Collections(t *testing.T) {
+	header := mocks.GenericHeader
+	blockData := mocks.GenericBlockData()
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(header.Height, header.ID())))
+
+		holder := mocks.BaselineRecordHolder(t)
+		holder.RecordFunc = func(blockID flow.Identifier) (*uploader.BlockData, error) {
+			assert.Equal(t, header.ID(), blockID)
+
+			return blockData, nil
+		}
+
+		cons := tracker.BaselineConsensus(
+			t,
+			tracker.WithDB(db),
+			tracker.WithLast(header.Height),
+			tracker.WithHolder(holder),
+		)
+
+		got, err := cons.Collections(header.Height)
+
+		require.NoError(t, err)
+		require.Len(t, got, len(blockData.Collections))
+		for idx, collection := range blockData.Collections {
+			for _, tx := range collection.Transactions {
+				assert.Contains(t, got[idx].Transactions, tx.ID())
+			}
+		}
```

Comment:
Unfortunately no because we are comparing `got` which contains ALL transactions against multiple slices of transactions:

```
Collection 1: 1, 2, 3, 4
Collection 2: 5, 6, 7, 8

Got: 1, 2, 3, 4, 5, 6, 7, 8
```

So `assert.ElementsMatch` would not work here. I originally thought I could do `assert.Contains(t, got, collection.Transactions)` and that it would try to match the contents of the slice as a subset of the other slice, but alas, it does not. So I don't think we can really do better than this.